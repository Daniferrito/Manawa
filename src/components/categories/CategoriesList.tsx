import React from 'react';
import Category from '../../types/Category';
import CategoriesListItem from './CategoriesListItem';

import ExampleData from '../../dummies/ExampleData.json';
import MainData from '../../types/MainData';

interface Props {
  categories?: Category[];
}

const CategoriesList = (props: Props) => {
  const { categories = MainData.fromJson(ExampleData).categories } = props;

  // if (categories === undefined) return <>{'Error'}</>;

  return (
    <>
      {categories.map(category => (
        <CategoriesListItem key={category.title} category={category} />
      ))}
    </>
  );
};

export default CategoriesList;
