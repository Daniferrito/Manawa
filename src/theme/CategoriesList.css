.container{
    display: grid;
    grid-template-columns: 50% 50%;
}

.title{
    grid-column-start: 1;
    grid-column-end: -1;
}

.subElement{
    grid-column: auto;
}