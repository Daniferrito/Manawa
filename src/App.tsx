import React, { Suspense } from 'react';
import { Redirect, Route } from 'react-router-dom';
import { IonApp, IonRouterOutlet } from '@ionic/react';
import { IonReactRouter } from '@ionic/react-router';
import CategoriesList from './components/categories/CategoriesList';
import Loader from './components/miscellaneous/Loader';

/* Core CSS required for Ionic components to work properly */
import '@ionic/react/css/core.css';

/* Basic CSS for apps built with Ionic */
import '@ionic/react/css/normalize.css';
import '@ionic/react/css/structure.css';
import '@ionic/react/css/typography.css';

/* Optional CSS utils that can be commented out */
import '@ionic/react/css/padding.css';
import '@ionic/react/css/float-elements.css';
import '@ionic/react/css/text-alignment.css';
import '@ionic/react/css/text-transformation.css';
import '@ionic/react/css/flex-utils.css';
import '@ionic/react/css/display.css';

/* Theme variables */
import './theme/variables.css';
import './theme/CategoriesList.css';

const App: React.FC = () => (
  <Suspense fallback={<Loader />}>
    <IonApp>
      <IonReactRouter>
        <IonRouterOutlet>
          <Route path="/home" component={CategoriesList} />
          <Route exact path="/" component={() => <Redirect to="/home" />} />
        </IonRouterOutlet>
      </IonReactRouter>
    </IonApp>
  </Suspense>
);

export default App;
