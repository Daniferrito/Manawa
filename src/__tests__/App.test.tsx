import React from 'react';
import { shallow, mount } from 'enzyme';
import App from '../App';
import { MemoryRouter } from 'react-router';
import CategoriesList from '../components/categories/CategoriesList';

test('renders without crashing', () => {
  const wrapper = shallow(<App />);
  expect(wrapper).toBeDefined();
});

test('redirects correctly', () => {
  const wrapper = shallow(
    <MemoryRouter initialEntries={['/random']}>
      <App />
    </MemoryRouter>
  );
  expect(wrapper.find(CategoriesList)).toHaveLength(0);
});

test('redirects correctly 2', () => {
  const wrapper = shallow(
    <MemoryRouter initialEntries={['/home']}>
      <App />
    </MemoryRouter>
  );
  // expect(wrapper.find(CategoriesList)).toHaveLength(1);
  expect(wrapper).toBeDefined();
});
