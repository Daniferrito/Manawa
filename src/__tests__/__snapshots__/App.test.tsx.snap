// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`redirects correctly 2 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "canGo": [Function],
      "createHref": [Function],
      "entries": Array [
        Object {
          "hash": "",
          "key": "15ls1o",
          "pathname": "/home",
          "search": "",
          "state": undefined,
        },
      ],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "index": 0,
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "key": "15ls1o",
        "pathname": "/home",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <App />
</Router>
`;

exports[`renders without crashing 1`] = `
<Suspense
  fallback={<Loader />}
>
  <ForwardRef(IonApp)>
    <IonReactRouter>
      <ForwardRef(IonRouterOutlet)>
        <Route
          component={[Function]}
          path="/home"
        />
        <Route
          component={[Function]}
          exact={true}
          path="/"
        />
      </ForwardRef(IonRouterOutlet)>
    </IonReactRouter>
  </ForwardRef(IonApp)>
</Suspense>
`;
